docker/vernemq/bin/vernemq.sh

```diff
@@ -0,0 +1,155 @@
+    for kube_pod_name in $kube_pod_names;
+    do
+        if [ $kube_pod_name == "null" ]
+            then
+                echo "Kubernetes discovery selected, but no pods found. Maybe we're the first?"
+                echo "Anyway, we won't attempt to join any cluster."
+                break
+        fi
+        if [ $kube_pod_name != $MY_POD_NAME ]
+            then
+                echo "Will join an existing Kubernetes cluster with discovery node at ${kube_pod_name}.${VERNEMQ_KUBERNETES_SUBDOMAIN}.${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE}.svc.${DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME}"
+                echo "-eval \"vmq_server_cmd:node_join('VerneMQ@${kube_pod_name}.${VERNEMQ_KUBERNETES_SUBDOMAIN}.${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE}.svc.${DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME}')\"" >> /vernemq/etc/vm.args
+                break
+        fi
+    done
+fi
+
+if [ -f /vernemq/etc/vernemq.conf.local ]; then
+    cp /vernemq/etc/vernemq.conf.local /vernemq/etc/vernemq.conf
+    sed -i -r "s/###IPADDRESS###/${IP_ADDRESS}/" /vernemq/etc/vernemq.conf
+else
+    sed -i '/########## Start ##########/,/########## End ##########/d' /vernemq/etc/vernemq.conf
+
+    echo "########## Start ##########" >> /vernemq/etc/vernemq.conf
+
+    env | grep DOCKER_VERNEMQ | grep -v 'DISCOVERY_NODE\|KUBERNETES\|SWARM\|COMPOSE\|DOCKER_VERNEMQ_USER' | cut -c 16- | awk '{match($0,/^[A-Z0-9_]*/)}{print tolower(substr($0,RSTART,RLENGTH)) substr($0,RLENGTH+1)}' | sed 's/__/./g' >> /vernemq/etc/vernemq.conf
+
+    users_are_set=$(env | grep DOCKER_VERNEMQ_USER)
+    if [ ! -z "$users_are_set" ]; then
+        echo "vmq_passwd.password_file = /vernemq/etc/vmq.passwd" >> /vernemq/etc/vernemq.conf
+        touch /vernemq/etc/vmq.passwd
+    fi
+
+    for vernemq_user in $(env | grep DOCKER_VERNEMQ_USER); do
+        username=$(echo $vernemq_user | awk -F '=' '{ print $1 }' | sed 's/DOCKER_VERNEMQ_USER_//g' | tr '[:upper:]' '[:lower:]')
+        password=$(echo $vernemq_user | awk -F '=' '{ print $2 }')
+        /vernemq/bin/vmq-passwd /vernemq/etc/vmq.passwd $username <<EOF
+$password
+$password
+EOF
+    done
+
+    echo "erlang.distribution.port_range.minimum = 9100" >> /vernemq/etc/vernemq.conf
+    echo "erlang.distribution.port_range.maximum = 9109" >> /vernemq/etc/vernemq.conf
+    echo "listener.tcp.default = ${IP_ADDRESS}:1883" >> /vernemq/etc/vernemq.conf
+    echo "listener.ws.default = ${IP_ADDRESS}:8080" >> /vernemq/etc/vernemq.conf
+    echo "listener.vmq.clustering = ${IP_ADDRESS}:44053" >> /vernemq/etc/vernemq.conf
+    echo "listener.http.metrics = ${IP_ADDRESS}:8888" >> /vernemq/etc/vernemq.conf
```

Comment:
You can turn off plugins.vmq_passwd here, it's not used
`echo "plugins.vmq_passwd = off" >> /vernemq/etc/vernemq.conf`

Reply:
This is true, but my goal here is to produce identical image as VerneMQ official image. Changing their scripts can be confusing, especially when we start investigating through a doc and we can not find what is changed where.

I propose that we stay aligned to their scripts and change explicitly in our k8s scripts via ENV vars - it is more documenting. Also, it permits us easier updates if their scripts change.